dbms/src/Parsers/formatAST.cpp

```diff
@@ -1,9 +1,23 @@
 #include <DB/Parsers/formatAST.h>
 
+#include <sstream>
 
 namespace DB
 {
 
+void formatAST(const IAST & ast, std::ostream & s, size_t indent, bool hilite, bool one_line)
+{
+	IAST::FormatSettings settings(s, hilite, one_line);
+	ast.format(settings);
+}
+
+std::string formatASTToString(const IAST & ast)
```

Comment:
Maybe, existing [queryToString](https://github.com/yandex/ClickHouse/blob/master/dbms/include/DB/Parsers/queryToString.h) will be better.